### ISSUE TITLE
expand macro recursively does nothing for ignored procmacros

### ISSUE THREAD

**Issue by tfreiberg-fastly** [rust-lang/rust-analyzer#14233]
If I improve goToReferences on function arguments by ignoring the `tracing-attributes::instrument` proc macro (as recommended in https://github.com/rust-lang/rust-analyzer/issues/12050), rust-analyzer won't expand that macro for me again.
The expansion buffer just shows the same function without the proc-macro attribute.

e.g.
```rust
#[tracing::instrument]
fn instrumented(s: String) {
    println!("Use s here: {}", s);
}
```
With the proc-macro ignored, goToReferences on `s` works, but the expanded buffer contains the following:

```rust
// Recursive expansion of instrument! macro
// =========================================

fn instrumented(s: String) {
    println!("Use s here: {}", s);
}
```

**rust-analyzer version**: rust-analyzer version: 0.3.1418-standalone (4e29820f6 2023-02-26)

**rustc version**: rustc 1.66.1 (90743e729 2023-01-10)

**relevant settings**: 
```json
    "rust-analyzer.procMacro.ignored": {
        "tracing-attributes": [
             "instrument"
        ]
    },
```

**Comment by lnicola** [comment]
That's what the setting does. It tells RA you don't want it to expand that macro.

**Comment by tfreiberg-fastly** [comment]
Yeah that's what I expected, but in my mind as a user, telling RA to ignore a macro during code analysis does not necessarily mean that I don't want it to help me view what the macro expansion looks like
I absolute don't need this changed, I just wanted to bring it up :)

**Comment by ChayimFriedman2** [comment]
Expanding macros not during analysis will entail complexities that I'm not sure are worth it.
